tools/yulPhaser/SimulationRNG.cpp

```diff
@@ -22,29 +22,35 @@
 #include <boost/random/uniform_int_distribution.hpp>
 
 #include <ctime>
+#include <limits>
 
+using namespace std;
 using namespace solidity;
 using namespace solidity::phaser;
 
 thread_local boost::random::mt19937 SimulationRNG::s_generator(SimulationRNG::generateSeed());
 
 bool SimulationRNG::bernoulliTrial(double _successProbability)
 {
-	boost::random::bernoulli_distribution<> distribution(_successProbability);
+	boost::random::bernoulli_distribution<double> distribution(_successProbability);
```

Comment:
`<double>` is default, do you prefer to have it explicit?

Reply:
I would vote for explicit `<double>`. Relying on default can make problem if default changes in different implementation/version.

Reply:
I made it explicit because I realized I had wrong assumptions about these types. For example I thought the default was `uint32_t` for the uniform int distribution but it was actually `int` (and I used `uint32_t` for parameters back then not becuause of it but because the mersenne twister accepts `uint32_t` and I wanted the types consistent - unnecessarily though).